ChainOfResponsibility.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace HelloWorld
+{
+    public interface IHandler
```

Comment:
again, the example and implementation is totally ok.
just an assumption the example in the article below could help you out with the better understanding of the pattern:
https://www.dofactory.com/net/chain-of-responsibility-design-pattern